### ISSUE TITLE
RCS1146 - false positive when there is an "and" condition

### ISSUE THREAD

**Issue by Arkane5** [dotnet/roslynator#1743]
RCS1146 is raised incorrectly in this configuration:
```cs
public void MyMethod(int[]? data)
{
    if (data != null && data.Length is not 2 and not 3) // RCS1146, suggesting: if (data?.Length is not 2 and not 3)
    {
        // data is not null, and its Length is not 2 or 3
        // After incorrect suggestion:
        // data is null or its Length is not 2 or 3
    }
    else
    {
        // data is null, or its Length is 2 or 3
        // After incorrect suggestion:
        // data is not null, its Length is 2 or 3
    }
}
```

This does not occur if there is no "and" condition (that is: `if (data != null && data.Length is not 2)`).

Workaround:
```cs
if (data != null && data is not { Length: 2 or 3 })
```

Related: #793
